src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -3182,19 +3182,12 @@ internal static bool TryGetStaticEntitySet(this IEdmPathExpression pathExpressio
 
         internal static bool HasAny<T>(this IEnumerable<T> enumerable) where T : class
         {
-            IList<T> list = enumerable as IList<T>;
-            if (list != null)
+            if (enumerable != null && enumerable.GetEnumerator().MoveNext())

```

Comment:
>&& enumerable.GetEnumerator().MoveNext()) [](start = 35, length = 41)

can you point me the materials to say to use this to test "HasAny"

Reply:
Refer to Line 804 in ExtensionMethodTests. 

---
In reply to: [362109873](https://github.com/OData/odata.net/pull/1602#discussion_r362109873) [](ancestors = 362109873)